pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -842,8 +842,8 @@ func (s *DiscoveryServer) updateEdsClients(added sets.Set, removed sets.Set, con
 			continue
 		}
 		c.mutex.Lock()
+		defer c.mutex.Unlock()
```

Comment:
This is not the cause, as L835 edsClusterMutex.Lock() can protect it. But i suggest fix it here too.

c.mutex.Lock()
delete(c.EdsClients, connection.ConID)
clients := len(c.EdsClients)
c.mutex.Unlock()

if clients == 0 {
...

Reply:
@hzxuzhonghu ok. Thanks. Fixed it. PTAL